src/snap/proxy/stream_collector_proxy.cc

```diff
@@ -0,0 +1,301 @@
+            }
+            std::this_thread::sleep_for(std::chrono::milliseconds(200));
+        }
+        _errChan.close();
+        _stream_collector->set_context_cancelled(true);
+        return true;
+    } catch (PluginException &e) {
+        std::cout << "Error" << std::endl;
+        return false;
+    }
+}
+
+bool StreamCollectorImpl::metricSend(const std::string &taskID,
+                                    ServerContext* context,
+                                    ServerReaderWriter<CollectReply, CollectArg>* stream) {
+    try {
+        std::chrono::system_clock::time_point start = std::chrono::system_clock::now();
+        while (!context->IsCancelled()) {
+            std::vector<Metric> send_mets;
+            if (_sendChan.get(send_mets)) {
+                if (!send_mets.empty()) {
+                    for (Metric met : send_mets) {
+                        *_metrics_reply.add_metrics() = *met.get_rpc_metric_ptr();
+                        if ((_metrics_reply.metrics_size() >= _max_metrics_buffer) && 
+                            _max_metrics_buffer != 0)  {
+                            //std::cout << "Max metrics buffer reached, sending metrics" << std::endl;
+                            sendReply(taskID, stream);
+                            _metrics_reply.clear_metrics();
+                            start = std::chrono::system_clock::now();                            
+                        }
+                    }
+                    if (_max_metrics_buffer == 0) {
+                        //std::cout << "Max metrics buffer set to 0, sending metrics" << std::endl;
+                        sendReply(taskID, stream);
+                        _metrics_reply.clear_metrics();
+                        start = std::chrono::system_clock::now();
+                    }
+                }
+            }
+
+            if ((std::chrono::system_clock::now() - start) >= _max_collect_duration) {
+                //std::cout << "Max collect duration reached, sending metrics" << std::endl;
+                sendReply(taskID, stream);
+                _metrics_reply.clear_metrics();
+               start = std::chrono::system_clock::now();
+            }
+        }
+        _sendChan.close();
+        _stream_collector->set_context_cancelled(true);
+        return true;
+    } catch (PluginException &e) {
+        std::cout << "Error" << std::endl;
+        return false;
+    }
+}
+
+bool StreamCollectorImpl::streamRecv(const std::string &taskID,
+                                    ServerContext* context,
+                                    ServerReaderWriter<CollectReply, CollectArg>* stream) {
+    try {
+        std::vector<Metric> recv_mets;
+        CollectArg collectMets;
+        while (!context->IsCancelled()) {
+            stream->Read(&collectMets);
+
+            if (collectMets.maxcollectduration() > 0) {
+                _max_collect_duration = std::chrono::seconds(collectMets.maxcollectduration());
```

Comment:
curretly max-collect-duration and max-metrics-buffer are placed in metrics configuration

Reply:
max-collect-duration and max-metrics-buffer currently initialize off the plugin->meta data, which gets the default values or the command line flag values. If user passes value in with task file, it updates here